desktop_version/src/Graphics.cpp

```diff
@@ -515,6 +515,118 @@ void Graphics::PrintAlpha( int _x, int _y, std::string _s, int r, int g, int b,
     }
 }
 
+bool Graphics::next_wrap(
+    size_t* start,
+    size_t* len,
+    const char* str,
+    const int maxwidth
+) {
+    /* This function is UTF-8 aware. But start/len still are bytes. */
+    size_t idx = 0;
+    size_t lenfromlastspace = 0;
+    size_t lastspace = 0;
+    int linewidth = 0;
+    *len = 0;
+
+    if (str[idx] == '\0')
+    {
+        return false;
+    }
+
+    while (true)
+    {
+        if ((str[idx] & 0xC0) == 0x80)
```

Comment:
This is not really future-proof and overall a bit of a "dirty" hack, will all non-ASCII characters always have the same width? My idea for future CJK localizations _would_ probably work with this since all characters (including ASCII) would just be 10x10 or 12x12, but that's not a good assumption to make since we _do_ have a dynamic `Graphics::bfontlen` and it takes a `uint32_t` for a reason.

Reply:
This would have to be changed later to actually concatenate the bytes together into a `uint32_t`, yes. The problem is I can't use UTF-CPP on C strings, so I would have to reimplement checking the bytes and concatenating them together (or find another library), and I don't need to do that now.

Reply:
This should be okay for 2.3, but will definitely need a FIXME so it can be found and addressed when necessary.

Reply:
Added a FIXME with a force-push.